pymatgen/io/lammps/sets.py

```diff
@@ -0,0 +1,219 @@
+        self.keep_stages = keep_stages
+
+        super().__init__(inputs={"in.lammps": self.inputfile, "system.data": self.data})
+
+    def get_inputs(self) -> dict[str, LammpsInputFile | LammpsData | CombinedData]:  # pylint: disable=E1131
+        """
+        Generate a dictionary of one or more input files to be written. Keys
+        are filenames, values are the contents of each file.
+        This method is called by write_input(), which performs the actual file
+        write operations.
+        """
+        return {"in.lammps": self.inputfile, "system.data": self.data}
+
+    @classmethod
+    def from_directory(cls, directory: str | Path, keep_stages: bool = False):  # pylint: disable=E1131
+        """
+        Construct a LammpsInputSet from a directory of two or more files.
+        TODO: accept directories with only the input file, that should include the structure as well.
+
+        Args:
+            directory: Directory to read input files from. It should contain at least two files:
+                       in.lammps for the LAMMPS input file, and system.data with the system information.
+            keep_stages: Whether to keep the stage structure of the LammpsInputFile or not.
+        """
+        input_file = LammpsInputFile.from_file(os.path.join(directory, "in.lammps"), keep_stages=keep_stages)
+        atom_style = input_file.get_args("atom_style")
+        data_file = LammpsData.from_file(os.path.join(directory, "system.data"), atom_style=atom_style)
+        return LammpsInputSet(inputfile=input_file, data=data_file, calc_type="read_from_dir")
+
+    def validate(self) -> bool:
+        """
+        A place to implement basic checks to verify the validity of an
+        input set. Can be as simple or as complex as desired.
+        Will raise a NotImplementedError unless overloaded by the inheriting class.
+        """
+        raise NotImplementedError(f".validate() has not been implemented in {self.__class__}")
+
+
+class BaseLammpsGenerator(InputGenerator):
+    r"""
+    Base class to generate LAMMPS input sets.
+    Uses template files for the input. The variables that can be changed
+    in the input template file are those starting with a $ sign, e.g., $nsteps.
+    This generic class is specialized for each template in subclasses, e.g. LammpsMinimization.
+    You can create a template for your own task following those present in pymatgen/io/lammps/templates.
+    The parameters are then replaced based on the values found
+    in the settings dictionary that you provide, e.g., `{"nsteps": 1000}`.
+
+    Parameters:
+        template: Path (string) to the template file used to create the input file for LAMMPS.
+        calc_type: String used to shortly describe the type of computations performed by LAMMPS.
+    Args:
+        settings: Dictionary containing the values of the parameters to replace in the template.
+        keep_stages: Whether to keep the stage structure of the LammpsInputFile or not.
```

Comment:
What is the difference between Parameters and Args? Also, it does not appear that `template` or `calc_type` can be set anywhere by the user (apologies if I'm missing it).

Reply:
I have refactored this part, thanks for pointing out that is was not consistent. Hopefully it's better now.

Reply:
Looks great, thank you!